### PR TITLE
[Help] Update links to new pattern

### PR DESCRIPTION
**Pull request by vincentsalucci**
## Type of change

- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Code changes
- **README.md**: Updated URL to new pattern

## Dependency
- [jslib - [Help] Update links to new pattern](https://github.com/bitwarden/jslib/pull/658)

## Testing requirements
- Make sure URL links don't 404 and take you to the correct page

## Before you submit

- [X] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ bwdc config --help
 
 **Detailed Documentation**
 
-We provide detailed documentation and examples for using the Directory Connector CLI in our help center at https://help.bitwarden.com/article/directory-sync/#command-line-interface.
+We provide detailed documentation and examples for using the Directory Connector CLI in our help center at https://bitwarden.com/help/directory-sync/#command-line-interface.
```
Comment (djsmith85) [inline README.md:45]: This link should probably be: https://bitwarden.com/help/directory-sync-cli/

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (vincentsalucci) [reply README.md:45]: @DanHillesheim @fschillingeriv - Thoughts? The redirect of the existing URL takes you to the URL I replaced it with. However, @djsmith85 makes a **great** point!

Comment (DanHillesheim) [reply README.md:45]: Thanks, that is a good point. I will leave that up to Fred though. I think he has final say on help site link destinations. @fschillingeriv What say ye?

Comment (fschillingeriv) [reply README.md:45]: Looks like this is specifically for BWDC CLI? In that case, yeah, it'd be best to link to https://bitwarden.com/help/directory-sync-cli/ !

Comment (vincentsalucci) [reply README.md]: Updated!

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**